src/main/java/io/jenkins/plugins/coverage/CoveragePublisher.java

```diff
@@ -0,0 +1,87 @@
+package io.jenkins.plugins.coverage;
+
+import hudson.DescriptorExtensionList;
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.model.AbstractProject;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.tasks.*;
+import io.jenkins.plugins.coverage.adapter.CoverageReportAdapter;
+import io.jenkins.plugins.coverage.adapter.CoverageReportAdapterDescriptor;
+import io.jenkins.plugins.coverage.targets.CoverageElement;
+import io.jenkins.plugins.coverage.targets.CoverageResult;
+import jenkins.tasks.SimpleBuildStep;
+import net.sf.json.JSONObject;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.StaplerRequest;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.util.List;
+
+public class CoveragePublisher extends Recorder implements SimpleBuildStep {
+
+    private CoverageReportAdapter[] adapters;
+
+    @DataBoundConstructor
+    public CoveragePublisher(CoverageReportAdapter[] adapters) {
+        this.adapters = adapters;
+    }
+
+    @Override
+    public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath workspace, @Nonnull Launcher launcher, @Nonnull TaskListener listener) throws InterruptedException, IOException {
+        CoverageProcessor processor = new CoverageProcessor();
+        List<CoverageResult> results = processor.precess(run, workspace, listener, adapters);
+
+        CoverageResult report = new CoverageResult(CoverageElement.AGGREGATED_REPORT, null, "All reports");
+        for (CoverageResult result : results) {
+            result.addParent(report);
+        }
+
+        CoverageAction action = new CoverageAction(report);
+        run.addAction(action);
+    }
+
+    @Override
+    public BuildStepMonitor getRequiredMonitorService() {
+        return BuildStepMonitor.NONE;
+    }
+
+
+    @Extension
+    public static final class CoveragePublisherDescriptor extends BuildStepDescriptor<Publisher> {
+
+        public CoveragePublisherDescriptor() {
+            super(CoveragePublisher.class);
+            load();
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public boolean configure(StaplerRequest req, JSONObject json) throws FormException {
+            save();
```

Comment:
why do invoke `save()` before super method? With such approach data on the disk may differ from the memory model